alacritty/src/config/ui_config.rs

```diff
@@ -85,70 +81,35 @@ pub struct UiConfig {
     /// Regex hints for interacting with terminal content.
     pub hints: Hints,
 
-    /// Offer IPC through a unix socket.
-    #[cfg(unix)]
-    pub ipc_socket: bool,
-
     /// Config for the alacritty_terminal itself.
     pub terminal: Terminal,
 
-    /// Path to a shell program to run on startup.
-    pub shell: Option<Program>,
-
-    /// Shell startup directory.
-    pub working_directory: Option<PathBuf>,
-
     /// Keyboard configuration.
     keyboard: Keyboard,
 
-    /// Should draw bold text with brighter colors instead of bold font.
-    #[config(deprecated = "use colors.draw_bold_text_with_bright_colors instead")]
-    draw_bold_text_with_bright_colors: bool,
-
-    /// Keybindings.
-    #[config(deprecated = "use keyboard.bindings instead")]
-    key_bindings: Option<KeyBindings>,
-
-    /// Bindings for the mouse.
-    #[config(deprecated = "use mouse.bindings instead")]
-    mouse_bindings: Option<MouseBindings>,
+    /// Path to a shell program to run on startup.
+    #[config(deprecated = "use terminal.shell instead")]
+    shell: Option<Program>,
 
     /// Configuration file imports.
     ///
     /// This is never read since the field is directly accessed through the config's
     /// [`toml::Value`], but still present to prevent unused field warnings.
-    import: Vec<String>,
-}
+    #[config(deprecated = "use general.import instead")]
+    import: Option<Vec<String>>,
 
-impl Default for UiConfig {
-    fn default() -> Self {
-        Self {
-            live_config_reload: true,
-            #[cfg(unix)]
-            ipc_socket: true,
-            draw_bold_text_with_bright_colors: Default::default(),
-            working_directory: Default::default(),
-            mouse_bindings: Default::default(),
-            config_paths: Default::default(),
-            key_bindings: Default::default(),
-            alt_send_esc: Default::default(),
-            scrolling: Default::default(),
-            selection: Default::default(),
-            keyboard: Default::default(),
-            terminal: Default::default(),
-            import: Default::default(),
-            cursor: Default::default(),
-            window: Default::default(),
-            colors: Default::default(),
-            shell: Default::default(),
-            mouse: Default::default(),
-            debug: Default::default(),
-            hints: Default::default(),
-            font: Default::default(),
-            bell: Default::default(),
-            env: Default::default(),
-        }
-    }
+    /// Shell startup directory.
+    #[config(deprecated = "use general.working_directory instead")]
+    working_directory: Option<PathBuf>,
+
+    /// Live config reload.
+    #[config(deprecated = "use general.live_config_reload instead")]
+    live_config_reload: Option<bool>,
+
+    /// Offer IPC through a unix socket.
+    #[cfg(unix)]
```

Comment:
Maybe `allow(unused)` here as well?

Reply:
I intentionally didn't change it here because it's deprecated anyway. If you have problems with this warning it's just one more reason to migrate away.

Reply:
yeah, I don't mind much for deprecated that much.